experimental/InjectiveResolutions/src/ModuleFunctionality.jl

```diff
@@ -0,0 +1,301 @@
+  M = ModuleGens(M.O, M.F, ordering)
+  mg, mat = lift_std(M)
+  mg.ordering = ordering
+  return mg, mat
+end
+
+function sparse_row(
+    A::MonoidAlgebra{<:FieldElem, <:MPolyRing}, 
+    svec::Singular.svector, rng::AbstractUnitRange
+  )
+  pre_res = sparse_row(A.algebra, svec, rng)
+  return map_entries(A, pre_res)
+end
+
+function sparse_row(
+    A::MonoidAlgebra{<:FieldElem, <:MPolyQuoRing}, 
+    svec::Singular.svector, rng::AbstractUnitRange
+  )
+  pre_res = sparse_row(base_ring(A.algebra), svec, rng)
+  return map_entries(A, pre_res)
+end
+
+function syzygy_module(F::ModuleGens{T}; sub = FreeMod(base_ring(F.F), length(oscar_generators(F)))) where {T <: MonoidAlgebraElem}
+  singular_assure(F)
+  # TODO Obtain the Gröbner basis and cache it
+  s = Singular.syz(singular_generators(F))
+  return SubquoModule(sub, s)
+end
+
+function kernel(
+    h::FreeModuleHom{<:FreeMod{T}, <:FreeMod{T}, Nothing}
+  ) where {S<:FieldElem, T <: MonoidAlgebraElem{S}}
+  is_zero(h) && return sub(domain(h), gens(domain(h)))
+  is_graded(h) && return _graded_kernel(h)
+  return _simple_kernel(h)
+end
+
+### Additional adjustments to get the graded aspects to run
+function annihilator(N::SubquoModule{T}) where T <: MonoidAlgebraElem
+  R = base_ring(N)
+  N_quo = isdefined(N, :quo) ? N.quo : SubModuleOfFreeModule(ambient_free_module(N), Vector{elem_type(ambient_free_module(N))}())
+  A = N.sub
+  SA = singular_generators(A.gens) 
+  B = N_quo
+  SB = singular_generators(B.gens)
+  res = Singular.quotient(SB, SA)
+  return ideal(R, [R(f) for f in gens(res)])
+end
+
+function singular_generators(J::MonoidAlgebraIdeal)
+  return singular_generators(underlying_ideal(J).gens)
+end
+
+function degree(v::FreeModElem{T}; check::Bool=true) where {T <: MonoidAlgebraElem}
+  is_zero(coordinates(v)) && return zero(grading_group(parent(v)))
+  @check is_homogeneous(v) "element is not homogeneous"
+  i, c = first(coordinates(v))
+  return parent(v).d[i] + degree(c)
+end
+
+# `twist` stuff is only copy-pasted because the original signature was too narrow.
+# This should be streamlined eventually.
+function twist(M::ModuleFP{T}, g::FinGenAbGroupElem) where {T <: MonoidAlgebraElem}
+ error("Not implemented for the given type")
+end
+
+function twist(M::SubquoModule{T}, g::FinGenAbGroupElem) where {T<:MonoidAlgebraElem}
+ R = base_ring(M)
+ @req parent(g) == grading_group(R) "Group element not contained in grading group of base ring"
+ F = ambient_free_module(M)
+ FN = twist(F, g)
+ shift = hom(F, FN, gens(FN); check=false)
+ return SubquoModule(FN, shift.(ambient_representatives_generators(M)), shift.(relations(M)))
+ # The original code below was giving strange errors from somewhere in AA.
+ GN = free_module(R, ngens(M))
+ HN = free_module(R, length(relations(M)))
+ a = hom(GN, F, ambient_representatives_generators(M); check=false)
+ b = hom(HN, F, relations(M); check=false)
+ A = matrix(a)
+ B = matrix(b)
+ N = subquotient(FN, A, B)
+ return N
```

Comment:
I'd recommend to either not keep this, or keep it commented out, else it'll be very easy to miss the early abort and be confused when studying the code.
```suggestion
```

Of course it would also be nice to know what the "strange errors" where and to debug them in AA but that's a separate issue of course :-)